tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/rules/MuleDeployment.java

```diff
@@ -88,11 +90,42 @@ public Builder withProperty(String property, String value)
             deployment.properties.put(property, value);
             return this;
         }
+
+        public Builder withApplication(String application)
+        {
+            deployment.applications.add(application);
+            return this;
+        }
+
+        public Builder withApplications(List<String> applications)
+        {
+            for (String application : applications)
+            {
+                deployment.applications.add(application);
+            }
+            return this;
+        }
+
+        public Builder withDomain(String domain)
+        {
+            deployment.domains.add(domain);
+            return this;
+        }
+
+        public Builder withDomains(List<String> domains)
```

Comment:
Use withDomains(String...domains) and consider removing withDomain(String domain)